src/main/java/org/opensearch/ad/rest/RestStatsAnomalyDetectorAction.java

```diff
@@ -11,64 +11,150 @@
 
 package org.opensearch.ad.rest;
 
-import static org.opensearch.ad.AnomalyDetectorPlugin.AD_BASE_URI;
-import static org.opensearch.ad.AnomalyDetectorPlugin.LEGACY_AD_BASE;
+import static org.opensearch.ad.util.RestHandlerUtils.NODE_ID;
+import static org.opensearch.ad.util.RestHandlerUtils.STAT;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Locale;
 import java.util.Set;
 import java.util.TreeSet;
-
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.ad.AnomalyDetectorExtension;
 import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
 import org.opensearch.ad.stats.ADStats;
 import org.opensearch.ad.transport.ADStatsRequest;
 import org.opensearch.ad.transport.StatsAnomalyDetectorAction;
+import org.opensearch.ad.transport.StatsAnomalyDetectorResponse;
 import org.opensearch.ad.util.DiscoveryNodeFilterer;
-import org.opensearch.client.node.NodeClient;
+import org.opensearch.ad.util.RestHandlerUtils;
 import org.opensearch.cluster.node.DiscoveryNode;
-import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.common.Strings;
-import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestRequest;
-import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.RouteHandler;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.sdk.SDKClusterService;
 
 import com.google.common.collect.ImmutableList;
 
 /**
- * RestStatsAnomalyDetectorAction consists of the REST handler to get the stats from the anomaly detector plugin.
+ * RestStatsAnomalyDetectorAction consists of the REST handler to get the stats from the anomaly detector extension.
  */
-public class RestStatsAnomalyDetectorAction extends BaseRestHandler {
+public class RestStatsAnomalyDetectorAction extends AbstractAnomalyDetectorAction {
```

Comment:
why extend the `AbstractAnomalyDetectorAction` instead of `BaseExtensionRestHandler`

Reply:
Looks like it gets that base class by inheritance, and adds access to several settings in the process that will get updated via settings update consumers.

However, later on in this class we're just fetching the setting directly and not taking advantage of the inheritance.

So either way is good but we should be consistent!

Reply:
Okay. I am replacing it with BaseExtensionRestHandler to maintain the consistency